gitlab/client.py

```diff
@@ -256,6 +258,87 @@ def from_config(
             retry_transient_errors=config.retry_transient_errors,
         )
 
+    @classmethod
+    def merge_config(
+        cls,
+        options: Namespace,
+        gitlab_id: Optional[str] = None,
+        config_files: Optional[List[str]] = None,
+    ) -> "Gitlab":
+        """Create a Gitlab connection by merging configuration with
+        the following precedence:
+
+        1. Explicitly provided CLI arguments,
+        2. Environment variables,
+        3. Configuration files:
+            a. explicitly defined config files:
+                i. via the `--config-file` CLI argument,
+                ii. via the `PYTHON_GITLAB_CFG` environment variable,
+            b. user-specific config file,
+            c. system-level config file,
+        4. Environment variables always present in CI (CI_SERVER_URL, CI_JOB_TOKEN).
+
+        Args:
+            options list[str]: List of options provided via the CLI.
+            gitlab_id (str): ID of the configuration section.
+            config_files list[str]: List of paths to configuration files.
+        Returns:
+            (gitlab.Gitlab): A Gitlab connection.
+
+        Raises:
+            gitlab.config.GitlabDataError: If the configuration is not correct.
+        """
+        config = gitlab.config.GitlabConfigParser(
+            gitlab_id=gitlab_id, config_files=config_files
+        )
+        url = (
+            options.server_url
+            or config.url
+            or os.getenv("CI_SERVER_URL")
+            or gitlab.const.DEFAULT_URL
+        )
+        private_token, oauth_token, job_token = cls._get_auth_from_env(options, config)
+
+        return cls(
+            url=url,
+            private_token=private_token,
+            oauth_token=oauth_token,
+            job_token=job_token,
+            ssl_verify=options.ssl_verify or config.ssl_verify,
+            timeout=options.timeout or config.timeout,
+            api_version=options.api_version or config.api_version,
+            per_page=options.per_page or config.per_page,
+            pagination=options.pagination or config.pagination,
+            order_by=options.order_by or config.order_by,
+            user_agent=options.user_agent or config.user_agent,
+        )
+
+    @staticmethod
+    def _get_auth_from_env(
```

Comment:
nit: I'm not sure about this name. mainly the 'from_env' part of it.

Reply:
True, especially with the comment above about removing `argparse` and using more generic dicts.

Reply:
I used `_merge_auth()` now, let me know if it's still confusing!